flake.nix

```diff
@@ -37,6 +37,30 @@
     {
       overlays.default = final: prev: {
         moduleit = self.packages.${prev.system}.moduleit;
+
+        lib = prev.lib // {
+          mkWrapper-replit_ld_library_path = package: final.stdenvNoCC.mkDerivation {
+            name = "${package.name}-wrapped";
+            inherit (package) meta version;
+
+            buildInputs = [ pkgs.makeWrapper ];
+            buildCommand = ''
+                            mkdir -p $out/bin
+
+                            for bin in ${package}/bin/*; do
```

Comment:
so, one annoyance is that if there _is_ no `bin` in a package, this'll just be the literal string specified here:
<img width="680" alt="image" src="https://github.com/replit/nixmodules/assets/278900/9cc45897-f439-4331-b8b5-452900405de0">

probably not a huge deal, but considering you're learning shell, it seems prudent to mention.

Maybe throw in a `[ ! -d ${package}/bin ] || exit 0`, or a `bins=( ${package}/bin/* )`/`[ -f ${bins[0]} ] || exit 0`/`for bin in "${bins[@]}"; do`

Reply:
🙏